packages/teams-js/test/private/teams.spec.ts

```diff
@@ -1,90 +1,154 @@
-          channelType: teams.ChannelType.Regular,
-        },
-      ];
-
-      const callback = jest.fn((err, folders) => {
-        expect(err).toBeFalsy();
-        expect(folders).toEqual(mockTeamsChannels);
+    describe('Testing teams.refreshSiteUrl function', () => {
+      it('teams.refreshSiteUrl should not allow calls before initialization', () => {
+        expect(() => teams.refreshSiteUrl('threadId', emptyCallback)).toThrowError(
+          'The library has not yet been initialized',
+        );
       });
 
-      teams.getTeamChannels('groupId', callback);
+      Object.values(FrameContexts).forEach(context => {
+        it(`teams.refreshSiteUrl should throw error when teams is not supported. context: ${context}`, async () => {
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 1, supports: {} });
+          expect.assertions(1);
+          try {
+            teams.refreshSiteUrl('threadId', emptyCallback);
+          } catch (e) {
+            expect(e).toEqual(errorNotSupportedOnPlatform);
+          }
+        });
 
-      const getCloudStorageFoldersMessage = utils.findMessageByFunc('teams.getTeamChannels');
-      expect(getCloudStorageFoldersMessage).not.toBeNull();
-      utils.respondToMessage(getCloudStorageFoldersMessage, false, mockTeamsChannels);
-      expect(callback).toHaveBeenCalled();
-    });
-  });
+        it(`teams.refreshSiteUrl should throw error when threadId is null. context: ${context}`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => teams.refreshSiteUrl('', emptyCallback)).toThrowError(
+            '[teams.refreshSiteUrl] threadId cannot be null or empty',
+          );
+        });
+
+        it(`teams.refreshSiteUrl should throw error when threadId is null. context: ${context}`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => teams.refreshSiteUrl('threadId', null)).toThrowError(
+            '[teams.refreshSiteUrl] Callback cannot be null',
+          );
+        });
+
+        it(`teams.refreshSiteUrl should trigger callback correctly. context: ${context}`, async () => {
+          await utils.initializeWithContext(context);
+          const callback = jest.fn(err => {
+            expect(err).toBeFalsy();
+          });
 
-  describe('refreshSiteUrl', () => {
-    it('should not allow calls before initialization', () => {
-      expect(() => teams.refreshSiteUrl('threadId', emptyCallback)).toThrowError(
-        'The library has not yet been initialized',
-      );
+          teams.refreshSiteUrl('threadId', callback);
+
+          const getCloudStorageFoldersMessage = utils.findMessageByFunc('teams.refreshSiteUrl');
+          expect(getCloudStorageFoldersMessage).not.toBeNull();
+          utils.respondToMessage(getCloudStorageFoldersMessage, false, 'Some Message');
+          expect(callback).toHaveBeenCalled();
+        });
+      });
```

Comment:
Thanks for cleaning up the tests and making it clearer what work is left to do (e.g., add frameless scenarios, etc.). Would it be possible to add tests for `isSupported`?  Since this PR is technically about ensuring `isSupported` is called, it seems appropriate to have these tests as part of this PR.